pkg/csi/plugins/nodeserver.go

```diff
@@ -313,14 +315,31 @@ func (ns *nodeServer) getRuntimeNamespacedName(volumeContext map[string]string,
 		runtimeName, nameFound := volumeContext[common.VolumeAttrName]
 		runtimeNamespace, nsFound := volumeContext[common.VolumeAttrNamespace]
 		if nameFound && nsFound {
+			glog.Infof("Get runtime namespace(%s) and name(%s) from volume context", runtimeNamespace, runtimeName)
 			return runtimeNamespace, runtimeName, nil
 		}
 	}
 
 	// Fallback: query API Server to get namespaced name
+	glog.Infof("Get runtime namespace and name directly from api server with volumeId %s", volumeId)
 	return volume.GetNamespacedNameByVolumeId(ns.apiReader, volumeId)
 }
 
+// getNode first checks cached node
+func (ns *nodeServer) getNode() (node *v1.Node, err error) {
+	if ns.node != nil {
```

Comment:
I think we need an environment variable to determine if we should use cache for node.

Reply:
Done